src/components/Button/Button.tsx

```diff
@@ -170,4 +176,8 @@ const Button: ButtonType = ({
 
 Button.Addon = Addon;
 
+Button.WithRef = React.forwardRef((props, ref) => (
```

Comment:
This addresses issue #291, in addition to helping me trigger focus on the current selected option when a `Select` is opened.